pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,68 @@
+package agent
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	// Cluster name
+	Name string `json:"name"`
+	// Installation will start once cluster is in known state and approved
+	Approved                 bool   `json:"approved,omitempty"`
```

Comment:
Approval feels imperative to me, if the ClusterDeployment has been created, the request has been made to create a cluster. We might be waiting in the controllers until some minimum number of machines is present. 

Propose we drop this at least for now, and assume it's approved in the controller code.

Reply:
I think this field delegates responsibility to the user for waiting until all of the expected agents have checked in and completed their inspection and validation. We otherwise do not (yet!) have a way to declare "wait until 3 agents are available with the role 'master' and 5 with the role 'worker'" or similar. Perhaps we should add that now, because it definitely matches the declarative pattern better.

Reply:
This is actually required for the interactive discovery flow, where the user doesn't know details about the hosts that will boot, wants to see validation results, wants to see default values that the service chose and the user might want to override.  There is also a security concern in BIY mode, where we don't automatically want to add every host that boots the ISO without the user's approval.  It's an imperative process, there's no way around it as far as I can tell.

Reply:
Sounds similar to CertificateSigningRequest.

Feels like Clusters are not what is actually getting approved, rather Agents coming up and trying to bind to a cluster. I would pitch that we move this off cluster and instead add the min requirements to start installation. Then the Agent API could optionally require approval to count against those min requirements.

Reply:
We will replace with a structure that defines minimum required control plane hosts (1 or 3), and a label selector to associate Agents, and same for workers. Agent CRD will have a role field, so the label selector can be used for both.